### ISSUE TITLE
Cannot install github-linguist.

### ISSUE THREAD

**Issue by gluons** [oneclick/rubyinstaller2#73]
This project is for Ruby version 2.4 and newer.
For Ruby versions < 2.4 please file an issue [here]( https://github.com/oneclick/rubyinstaller/issues/new ).

## What problems are you experiencing?
Cannot install [`github-linguist`](https://github.com/github/linguist).

## Steps to reproduce
```bash
> gem install github-linguist

Temporarily enhancing PATH for MSYS/MINGW...
Building native extensions.  This could take a while...
ERROR:  Error installing github-linguist:
        ERROR: Failed to build gem native extension.

***************************************************************************************
*********** icu required (brew install icu4c or apt-get install libicu-dev) ***********
***************************************************************************************
*** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.
```

## What's the output from `ridk version`?
```yml
---
ruby:
  path: C:/Ruby24-x64
  version: 2.4.1
  platform: x64-mingw32
ruby_installer:
  package_version: 2.4.1-2
  git_commit: a2667ee
msys2:
  path: C:\msys64
  title: MSYS2 64bit
  version: '20161025'
cc: x86_64-w64-mingw32-gcc (Rev2, Built by MSYS2 project) 7.1.0
sh: GNU bash, version 4.4.12(1)-release (x86_64-pc-msys)
```

**Comment by gluons** [comment]
**Related:** github/linguist#3649

**Comment by MSP-Greg** [comment]
@gluons

See my response over at the link you mentioned.  Please let us know if that allowed install...

Summary -
1. Install the package icu
2. Copy two files

**Comment by gluons** [comment]
OK. I'll try.

**Comment by gluons** [comment]
**New problem:**

```
> gem install github-linguist

Temporarily enhancing PATH for MSYS/MINGW...
Building native extensions.  This could take a while...
Successfully installed charlock_holmes-0.7.5
Fetching: escape_utils-1.1.1.gem (100%)
Building native extensions.  This could take a while...
ERROR:  Error installing github-linguist:
        ERROR: Failed to build gem native extension.

    current directory: C:/Ruby24-x64/lib/ruby/gems/2.4.0/gems/escape_utils-1.1.1/ext/escape_utils
C:/Ruby24-x64/bin/ruby.exe -r ./siteconf20170828-18620-16wagsx.rb extconf.rb
creating Makefile

current directory: C:/Ruby24-x64/lib/ruby/gems/2.4.0/gems/escape_utils-1.1.1/ext/escape_utils
make "DESTDIR=" clean

current directory: C:/Ruby24-x64/lib/ruby/gems/2.4.0/gems/escape_utils-1.1.1/ext/escape_utils
make "DESTDIR="
generating escape_utils-x64-mingw32.def
compiling buffer.c
buffer.c: In function 'gh_buf_cmp':
buffer.c:245:38: warning: implicit declaration of function 'MIN' [-Wimplicit-function-declaration]
  int result = memcmp(a->ptr, b->ptr, MIN(a->size, b->size));
                                      ^~~
compiling escape_utils.c
compiling houdini_href_e.c
compiling houdini_html_e.c
compiling houdini_html_u.c
compiling houdini_js_e.c
compiling houdini_js_u.c
compiling houdini_uri_e.c
compiling houdini_uri_u.c
compiling houdini_xml_e.c
linking shared-object escape_utils/escape_utils.so
buffer.o:buffer.c:(.text+0x5b5): undefined reference to `MIN'
collect2.exe: error: ld returned 1 exit status
make: *** [Makefile:259: escape_utils.so] Error 1

make failed, exit code 2
```

**Comment by MSP-Greg** [comment]
@gluons 

Sorry, I just looked at the dependencies, and saw the first one (charlock_holmes) was an extension.  Assumed that would allow github-linguist to install.  My mistake for not checking further.

The problem you've listed might be a bit worse.  I'm offline soon, I'll try to look at it tomorrow.

@brianmario is involved with both charlock_holmes and escape_utils, maybe he'll respond.  Re the charlock_holmes issue with differing file names, ruby/opensll has code to check for different files, as the files for OpenSSL 1.0.2 and 1.1.0 are different.  He may be able to patch charlock_holmes so a symlink/copy isn't needed.  Don't know about escape_utils...

**Comment by brianmario** [comment]
👋 It looks like you got the charlock_holmes issue was resolved, and the issue with `escape_utils` was [fixed in 1.2.0](https://github.com/brianmario/escape_utils/commit/3373cd6173846a9708b2f368679002a06e8ab909). How hard would it be to bump that dependency?

**Comment by larskanis** [comment]
Thank you for your contributions! However this is no longer a RubyInstaller issue. With the MSYS2 foundation as our base, all required dependencies ( [icu4c](https://github.com/Alexpux/MINGW-packages/tree/master/mingw-w64-icu), `cmake` and `pkg-config`) can easily be installed per pacman or per [gem dependencies](https://github.com/oneclick/rubyinstaller2/wiki/For-gem-developers#user-content-msys2-library-dependency). It's therefore up to the dependent gems to improve their Windows support and fix the remaining issues.
